### PR TITLE
Line wrap description comments

### PR DESCRIPTION
**Pull request by pongad**
Updates #184.

### CODE REVIEW

File: src/main/java/com/google/api/codegen/DiscoveryContext.java [hunk 1]
```diff
@@ -162,6 +163,41 @@ public boolean isPatch(Method method) {
     return apiaryConfig.getHttpMethod(method.getName()).equals("PATCH");
   }
 
+  // line wrap `str`, returning a list of lines. Each line in the returned list is guaranteed to
```
Comment (broady) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: JavaDoc comments should be complete sentences.

File: src/main/java/com/google/api/codegen/DiscoveryContext.java [hunk 2]
```diff
@@ -162,6 +163,41 @@ public boolean isPatch(Method method) {
...
+  // not have new line characters.
+  public List<String> lineWrap(String str, int col) {
```
Comment (broady) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: s/col/maxWidth/

File: src/main/java/com/google/api/codegen/DiscoveryContext.java [hunk 3]
```diff
@@ -162,6 +163,41 @@ public boolean isPatch(Method method) {
...
+        for (int i = 0; i < str.length(); i++) {
+          if (Character.isWhitespace(str.charAt(i))) {
+            splitPos = i;
+            break;
+          }
+        }
+      }
+      if (splitPos < 0) {
+        lines.add(str.trim());
+        str = "";
```
Comment (broady) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: break, don't set str = ""

**Comment by broady** [comment]
LGTM

File: src/main/java/com/google/api/codegen/DiscoveryContext.java (hunk 1 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: Uppercased 'l'

File: src/main/java/com/google/api/codegen/DiscoveryContext.java (hunk 2 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: Done

File: src/main/java/com/google/api/codegen/DiscoveryContext.java (hunk 3 above)
Comment (pongad) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: Done

File: src/main/java/com/google/api/codegen/DiscoveryContext.java [hunk 4]
```diff
@@ -162,6 +163,40 @@ public boolean isPatch(Method method) {
     return apiaryConfig.getHttpMethod(method.getName()).equals("PATCH");
   }
 
+  // Line wrap `str`, returning a list of lines. Each line in the returned list is guaranteed to
+  // not have new line characters.
+  public List<String> lineWrap(String str, int maxWidth) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: Maybe call this `lineWrapDoc` because it's not meant to line wrap code.

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

File: src/main/java/com/google/api/codegen/DiscoveryContext.java [hunk 5]
```diff
@@ -165,38 +165,43 @@ public boolean isPatch(Method method) {
 
   // Line wrap `str`, returning a list of lines. Each line in the returned list is guaranteed to
   // not have new line characters.
-  public List<String> lineWrap(String str, int maxWidth) {
+  public List<String> lineWrapDoc(String str, int maxWidth) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/DiscoveryContext.java]: Since this is essentially pure logic, I think we should have some unit tests.

**Comment by pongad** [comment]
@garrettjonesgoogle PTAL

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**